models/delay_and_sum_beamformer/sm_init_test_signals.m

```diff
@@ -0,0 +1,105 @@
+
+signal_option = 1;  % set which test signal to use
+
+switch signal_option
+    case 1 % Simple tones        
+        mp.test_signal.duration = 5;  % duration of tone in seconds
+        Nsamples = round(mp.test_signal.duration*mp.Fs);
+        if mp.fastsim_flag == 1 % perform fast simulation by reducing the number of samples
+           mp.test_signal.Nsamples = min(Nsamples, mp.fastsim_Nsamples);
+        else
+           mp.test_signal.Nsamples = mp.fastsim_Nsamples;
+        end
+        sample_times = [0 1:(mp.test_signal.Nsamples-1)]*mp.Ts;
+        mp.test_signal.data  = repmat(cos(2*pi*1000*sample_times), ... 
+            mp.arraySize(1)*mp.arraySize(2), 1);
+    case 2  % speech 
+        [y,Fs] = audioread('SpeechDFT-16-8-mono-5secs.wav');  % speech sample found in the Matlab Audio Toolbox 
+        y_resampled = resample(y,mp.Fs,Fs)';  % resample to change the sample rate to SG.Fs
+        Nsamples = length(y_resampled);
+        if mp.fastsim_flag == 1 % perform fast simulation by reducing the number of samples
+           mp.test_signal.Nsamples = min(Nsamples, mp.fastsim_Nsamples);
+        else
+           mp.test_signal.Nsamples = mp.fastsim_Nsamples;
+        end
+        mp.test_signal.data  = repmat(y_resampled(1:mp.test_signal.Nsamples), ...
+            mp.arraySize(1) * mp.arraySize(2), 1);
+        mp.test_signal.Nsamples = length(mp.test_signal.data);
+        mp.test_signal.duration = mp.test_signal.Nsamples * mp.Ts;
+    case 3 % square wave
+        mp.test_signal.duration = 5;  % duration of tone in seconds
+        Nsamples = round(mp.test_signal.duration*mp.Fs);
+        if mp.fastsim_flag == 1 % perform fast simulation by reducing the number of samples
+           mp.test_signal.Nsamples = min(Nsamples, mp.fastsim_Nsamples);
+        else
+           mp.test_signal.Nsamples = mp.fastsim_Nsamples;
+        end
+        sample_times = [0 1:(mp.test_signal.Nsamples-1)]*mp.Ts;
+        mp.test_signal.data  = repmat(square(2*pi*2000*sample_times), ... 
+            mp.arraySize(1)*mp.arraySize(2), 1);
+    otherwise
+        error('Please choose a viable option for the test signal (see sm_init_test_signals)')
+end
+
+%% simulate direction of arrival for each channel
+% the doa can be set arbitrarily between +/- 90 for both angles.
+% if you want the incoming doa to match the look direction of the array,
+% the angles here need to match those in the sm_init_control_signals.
+%
+% to simulate steering angle quantization errors due to fixed point data
+% types, use doubles for simulatedAzimuth and simulatedElevation and use 
+% the computeDelaysDoubles function; otherwise, use computeDelaysFixedPoint
+% sfix16_8 types for azimuth and elevation
+mp.simulatedAzimuth = 34;
+mp.simulatedElevation = -55;
+
+delays = computeDelays(mp);
+
+%plot(mp.test_signal.data(1,:)'); hold on;
+
+% delay the signal at each sensor
+mp.test_signal.data = delayseq(mp.test_signal.data', double(delays))';
```

Comment:
So this is what you wanted the phased array system toolbox for

Reply:
Yep! We should replace that at some point with our own function. Or just buy the toolbox :)

Reply:
I've found that you can find some open source functions on the MATLAB forums written by fellow software engineers. That might be a resource worth looking into as an alternative to buying the toolbox ;)

Reply:
I think I found those as well a few days ago 👍